src/Containers/Microsoft.NET.Build.Containers/Tasks/CreateNewImage.cs

```diff
@@ -240,4 +242,108 @@ public void Dispose()
     {
         _cancellationTokenSource.Dispose();
     }
+
+    internal (string[] entrypoint, string[] cmd) DetermineEntrypointAndCmd(string[]? baseImageEntrypoint)
+    {
+        string[] entrypoint = Entrypoint.Select(i => i.ItemSpec).ToArray();
+        string[] entrypointArgs = EntrypointArgs.Select(i => i.ItemSpec).ToArray();
+        string[] cmd = Cmd.Select(i => i.ItemSpec).ToArray();
+        string[] appCommand = AppCommand.Select(i => i.ItemSpec).ToArray();
+        string[] appCommandArgs = AppCommandArgs.Select(i => i.ItemSpec).ToArray();
+        string appCommandInstruction = AppCommandInstruction;
+
+        bool setsEntrypoint = entrypoint.Length > 0 || entrypointArgs.Length > 0;
+        bool setsCmd = cmd.Length > 0;
+
+        baseImageEntrypoint ??= Array.Empty<string>();
+        // Some (Microsoft) base images set 'dotnet' as the ENTRYPOINT. We mustn't use it.
+        if (baseImageEntrypoint.Length == 1 && (baseImageEntrypoint[0] == "dotnet" || baseImageEntrypoint[0] == "/usr/bin/dotnet"))
+        {
+            baseImageEntrypoint = Array.Empty<string>();
+        }
+
+        if (string.IsNullOrEmpty(appCommandInstruction))
+        {
+            if (setsEntrypoint)
+            {
+                // Backwards-compatibility: before 'AppCommand'/'Cmd' was added, only 'Entrypoint' was available.
+                if (!setsCmd && appCommandArgs.Length == 0 && entrypoint.Length == 0)
+                {
+                    // Copy over the values for starting the application from AppCommand.
+                    entrypoint = appCommand;
+                    appCommand = Array.Empty<string>();
+
+                    // Use EntrypointArgs as cmd.
+                    cmd = entrypointArgs;
+                    entrypointArgs = Array.Empty<string>();
+
+                    if (entrypointArgs.Length > 0)
+                    {
+                        // Log warning: Instead of ContainerEntrypointArgs, use ContainerAppCommandArgs for arguments that must always be set, or ContainerCmd for default arguments that the user override when creating the container.
+                        Log.LogWarningWithCodeFromResources(nameof(Strings.EntrypointArgsSetPreferAppCommandArgs));
+                    }
+
+                    appCommandInstruction = KnownAppCommandInstructions.None;
+                }
+                else
+                {
+                    // There's an Entrypoint. Use Cmd for the AppCommand.
+                    appCommandInstruction = KnownAppCommandInstructions.Cmd;
+                }
+            }
+            else
+            {
+                // Default to use an Entrypoint.
+                // If the base image defines an ENTRYPOINT, print a warning.
+                if (baseImageEntrypoint.Length > 0)
+                {
+                    Log.LogWarningWithCodeFromResources(nameof(Strings.BaseEntrypointOverwritten));
+                }
```

Comment:
Rather than change between `Entrypoint` and `Cmd` based on the presence of a base entrypoint, for deterministic behavior, I've opted to prefer `Entrypoint` and print a warning if the base image defines an entrypoint already (as is the case with Red Hat images).